From 5c17d45a049bca355ab228ebf5dd5c84d831aff9 Mon Sep 17 00:00:00 2001
From: Joachim Breitner <mail@joachim-breitner.de>
Date: Tue, 12 Jan 2010 11:45:37 +0100
Subject: Cast to (intptr_t) in xslt/impl/Instance.cxx

This is part of the debugging output.
Prevoius code did a check
((int)nsiContext().node() > 0x1000000)
to see whether the node name is valid. This looks fishy and
non-portable.
---
 sfworks/xslt/impl/Instance.cxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sfworks/xslt/impl/Instance.cxx b/sfworks/xslt/impl/Instance.cxx
index 4f04fb4..9813a47 100644
--- a/sfworks/xslt/impl/Instance.cxx
+++ b/sfworks/xslt/impl/Instance.cxx
@@ -343,7 +343,7 @@ void Instance::dump() const
         << " PS:" << prevSibling() << " NS:" << nextSibling()
         << " R" << getRefCnt() << " S#:" << siblingIndex_
         << " ctx "<< nsiContext().node()
-        << "/<" << (((int)nsiContext().node() > 0x1000000) ?
+        << "/<" << (((intptr_t)nsiContext().node() > 0x1000000) ?
             nsiContext().node()->nodeName() : String("*BAD*"))
         << ">" << std::endl;
     for (const Instance* i = firstChild(); i; i = i->nextSibling()) {
-- 
1.6.6

